### PR TITLE
Move creating of nav & menu items to legacycustomsearches extension

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Move creating of nav & menu items to legacycustomsearches extension

Before
----------------------------------------
Menu & navigation items for legacy custom searches declared in core code

After
----------------------------------------
Declared in the extension

Technical Details
----------------------------------------
As part of making it turn-offable....

Comments
----------------------------------------

### CODE REVIEW

File: ext/legacycustomsearches/legacycustomsearches.php [hunk 1]
```diff
@@ -95,14 +95,22 @@ function legacycustomsearches_civicrm_entityTypes(&$entityTypes) {
...
+  ]);
+  _legacycustomsearches_civix_insert_navigation_menu($menu, 'Administer/Customize Data and Screens', [
+    'label' => E::ts('Manage Custom Searches'),
+    'name' => 'Manage Custom Searches',
+    'url' => 'civicrm/admin/options/custom_search?reset=1',
+    'permission' => 'administer CiviCRM',
+    'operator' => 'OR',
+    'separator' => 0,
+  ]);
+  _legacycustomsearches_civix_navigationMenu($menu);
```
Comment (colemanw) [inline ext/legacycustomsearches/legacycustomsearches.php:115]: FYI. We have several ways for extensions to add to the nav menu, and none of them are very good:

1. Insert during installation (`hook_civicrm_install`) using the api
2. Use `hook_civicrm_navigationMenul`
3. 2. Use `hook_civicrm_managed`

They all have downsides. 1 requires manual cleanup during uninstall, and manual handling of multi-domains. 2 is non-configurable, frustrating admins who want to customize their menus. 3 works well as long as you take care of multi-domains yourself. So on balance I think 3 is my favorite, and I've recently started using this pattern:

https://github.com/civicrm/civicrm-core/blob/5bee23815cd56c5214634ee7f40a1b8f52f0ac20/ext/search_kit/managed/Navigation_search_kit.mgd.php

**Comment by colemanw** [comment]
@eileenmcnaughton this looks good except for the test failures :/

**Comment by eileenmcnaughton** [comment]
@colemanw I fixed the test failure that seemed to relate - hopefully this will pass now - would be good to get a MOP so I can merge if it does since it now contacts a `regen` for the sql. Note that the test fails were test-only issues

**Comment by colemanw** [comment]
@eileenmcnaughton still some test issues

**Pull request merged**